pkg/policies/dataplane/resources/fluxmeter/flux-meter.go

```diff
@@ -138,52 +139,26 @@ func (fluxMeterFactory *fluxMeterFactory) newFluxMeterOptions(
 }
 
 func (fluxMeter *FluxMeter) setup(lc fx.Lifecycle, prometheusRegistry *prometheus.Registry) {
-	metricLabels := make(map[string]string)
-	metricLabels[metrics.FluxMeterNameLabel] = fluxMeter.GetFluxMeterName()
-
 	lc.Append(fx.Hook{
 		OnStart: func(_ context.Context) error {
-			// Initialize a prometheus histogram metric
-			fluxMeter.histMetricVec = prometheus.NewHistogramVec(prometheus.HistogramOpts{
-				Name:        metrics.FluxMeterMetricName,
-				Buckets:     fluxMeter.buckets,
-				ConstLabels: prometheus.Labels{metrics.FluxMeterNameLabel: fluxMeter.fluxMeterName},
-			}, []string{
-				metrics.DecisionTypeLabel,
-				metrics.StatusCodeLabel,
-				metrics.FeatureStatusLabel,
-			})
-			// Register metric with Prometheus
-			err := prometheusRegistry.Register(fluxMeter.histMetricVec)
-			if err != nil {
-				log.Error().Err(err).Msgf("Failed to register metric %+v with Prometheus registry", fluxMeter.histMetricVec)
-				return err
-			}
-
 			// Register metric with PCA
-			err = engineAPI.RegisterFluxMeter(fluxMeter)
+			err := engineAPI.RegisterFluxMeter(fluxMeter)
 			if err != nil {
 				log.Error().Err(err).Msgf("Failed to register FluxMeter %s with EngineAPI", fluxMeter.fluxMeterName)
 				return err
 			}
 			return nil
 		},
 		OnStop: func(_ context.Context) error {
-			var errMulti error
 			// Unregister metric with PCA
 			err := engineAPI.UnregisterFluxMeter(fluxMeter)
 			if err != nil {
 				log.Error().Err(err).Msgf("Failed to unregister FluxMeter %s with EngineAPI", fluxMeter.fluxMeterName)
-				errMulti = multierr.Append(errMulti, err)
 			}
 
-			// Unregister metric with Prometheus
-			unregistered := prometheusRegistry.Unregister(fluxMeter.histMetricVec)
-			if !unregistered {
-				log.Error().Err(err).Msgf("Failed to unregister metric %+v with Prometheus registry", fluxMeter.histMetricVec)
-			}
```

Comment:
@DariaKunoichi,
Please restore this code. It is already properly handling the lifecycle for FluxMeter metrics. Need lifecylce handling for only the Workload metrics